src/AppInstallerCLICore/Workflows/PortableFlow.cpp

```diff
@@ -341,17 +545,23 @@ namespace AppInstaller::CLI::Workflow
                 AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PORTABLE_REPARSE_POINT_NOT_SUPPORTED);
             }
         }
-    }
+    }
 
     void PortableInstallImpl(Execution::Context& context)
     {
         try
         {
             context.Reporter.Info() << Resource::String::InstallFlowStartingPackageInstall << std::endl;
 
+            // Add product code for uninstall subcontext prior to install in case of install failure.
+            const std::string& productCode = GetPortableProductCode(context);

```

Comment:
Why not construct this here before the `try` then have the uninstall invoked from this function as well?  I don't see any value in splitting it into two tasks.

Reply:
Changed to constructing a PortableARPEntry before the `try` and adding that to context. Also changed to having the uninstall flow invoked from here